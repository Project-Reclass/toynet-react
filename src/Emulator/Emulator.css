.emulator {
  display: flex;
  max-width: 1920px;
  margin: auto;
}