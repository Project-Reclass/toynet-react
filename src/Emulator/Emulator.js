import React from 'react';
import Instructions from './Instructions/Instructions';
import Tabs from './Tabs/Tabs';
import Visuals from './Visuals/Visuals';

const Emulator = ({ panelData }) => {
  return (
    <div style={{display: 'flex'}}>
      <Instructions panelData={panelData} />
      <div>
        <Tabs />
        <Visuals />
      </div>
    </div>
  );
};

export default Emulator;

