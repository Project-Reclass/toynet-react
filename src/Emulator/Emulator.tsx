/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import React from 'react';
import { Grid } from '@chakra-ui/core';

import Visuals from './Visuals';
import Instructions from './Instructions';
import DialogueBox from './DialogueBox';
import Console from './Console';
import styled from '@emotion/styled-base';
import DrawerProvider from '../common/providers/DrawerProvider';

const data = {
  'id': 1,
  'submoduleNumber': 1,
  'submoduleName': 'Modifying Topology',
  'objective': 'Add and remove devices and their connections.',
  'tasks': [
    'Add a Host (h3)',
    'Attach h3 to s2',
    'Detach h3 from s2',
    'Remove h2',
  ],
};

const EmulatorGrid = styled(Grid)`
  margin: auto;
  max-width: 1920px;
  height: 100vh;
  max-height: 100vh;
  overflow: hidden;
  padding: 0.789rem;
  grid-template-columns: auto 1fr;
  @media (max-width: 2050px) {
    margin-left: 5rem;
  }
`;

const Emulator = () => {
  return (
    <DrawerProvider>
      <EmulatorGrid gap={2}>
        <Instructions panelData={data} />
        <Grid
          gridTemplateRows={'1fr 1fr'}
          width='100%'
          overflow='hidden'
          gap={2}
        >
          <Visuals />
          <Grid
            height='100%'
            maxH='100%'
            overflow='hidden'
            gap={2}
            gridTemplateColumns={'2fr 1fr'}
          >
            <Console />
            <DialogueBox />
          </Grid>
        </Grid>
      </EmulatorGrid>
    </DrawerProvider>
  );
};

export default Emulator;
