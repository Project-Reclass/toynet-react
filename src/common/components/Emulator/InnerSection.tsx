/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import { Box } from '@chakra-ui/react';
import styled from '@emotion/styled';

export const EmulatorInnerSection = styled(Box)`
  background-color: #212529;
  width: 100%;
  height: 100%;
  position: relative;
  flex: 1 1 auto;
  overflow-y: auto;
  padding: 0.549rem;
  margin-bottom: 0.549rem;
`;

export default EmulatorInnerSection;