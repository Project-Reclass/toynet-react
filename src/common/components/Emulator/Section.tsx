/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import { Box } from '@chakra-ui/react';
import styled from '@emotion/styled';

interface Props {
  width?: string;
  height?: string;
  minWidth?: string;
  maxWidth?: string;
  padding?: string
}

const EmulatorSection = styled(Box)`
  color: white;
  position: relative;
  background-color: #454950;
  height: ${(props: Props) => props.height || '100%'};
  width: ${(props: Props) => props.width || '100%'};
  border-radius: 10px;
  padding: ${(props: Props) => props.padding || '2vh'};
  display: flex;
  flex-direction: column;
  min-width: ${(props: Props) => props.minWidth};
  max-width: ${(props: Props) => props.maxWidth};
`;

export default EmulatorSection;