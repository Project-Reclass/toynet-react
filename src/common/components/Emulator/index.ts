export { default as EmulatorTitle } from './Title';
export { default as EmulatorInnerSection } from './InnerSection';
export { default as EmulatorSection } from './Section';