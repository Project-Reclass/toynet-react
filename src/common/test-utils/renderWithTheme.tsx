import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@chakra-ui/core/dist';
import renderer from 'react-test-renderer';

const Wrapper: FC = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

export const renderWithTheme = (ui: any) => {
  return render(ui, { wrapper: Wrapper });
};

export const renderTreeWithTheme = (ui: any) => {
  return renderer.create(<Wrapper>{ui}</Wrapper>);
};
