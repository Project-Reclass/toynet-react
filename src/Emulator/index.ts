import Emulator from './Emulator';
import { withEmulatorAndDialogueProvider } from './EmulatorProvider';

export default withEmulatorAndDialogueProvider(Emulator);