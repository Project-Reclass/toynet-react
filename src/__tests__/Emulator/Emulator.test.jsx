import React from 'react';
import renderer from 'react-test-renderer';
import { renderTreeWithTheme } from 'src/common/test-utils/renderWithTheme';

import Emulator from 'src/Emulator/Emulator';

describe('The emulator', () => {
  it('should render and match snapshots', () => {
    const tree = renderTreeWithTheme(<Emulator />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});