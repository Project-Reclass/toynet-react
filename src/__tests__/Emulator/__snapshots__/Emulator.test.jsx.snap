// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The emulator should render and match snapshots 1`] = `
<div
  className="emulator"
>
  <div
    className="instructions-background"
  >
    <div
      className="instructions"
    >
      <div
        className="instructions-exit"
      >
        <a
          className="svg"
          href="blank"
        >
          <svg>
            backIcon.svg
          </svg>
        </a>
        <a
          className="link-text"
          href="blank"
        >
          Back to course
        </a>
      </div>
      <div
        className="title-container"
      >
        <h4
          className="module-num"
        >
          Module 
          1
        </h4>
        <p
          className="module-name"
        >
          Local Area Networks
        </p>
      </div>
      <div
        className="objective-container"
      >
        <h4
          className="objective-title"
        >
          Objective
        </h4>
        <p
          className="objective"
        >
          Connect two hosts together and check they can communicate
        </p>
      </div>
      <div
        className="task-container"
      >
        <h4
          className="tasks-title"
        >
          Tasks
        </h4>
        <ol>
          <li
            className="steps"
          >
            Attach h1 to s1.
          </li>
          <li
            className="steps"
          >
            Attach h2 to s1.
          </li>
          <li
            className="steps"
          >
            Launch.
          </li>
          <li
            className="steps"
          >
            In Console, run h1&gt; arp 172.16.0.100.
          </li>
          <li
            className="steps"
          >
            In Console, run h1&gt; ping 172.16.0.100.
          </li>
          <li
            className="steps"
          >
            In Console, run h1&gt; arp 172.16.0.100.
          </li>
        </ol>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginLeft": "1.5vw",
        "marginTop": "2vh",
      }
    }
  >
    <div>
      <div
        className="tabs__container"
      >
        <div
          className="tabs__outer-tabs-container "
        >
          <div
            className="tabs__tabs-container "
          >
            <div
              className="tabs__tab tabs__tab--active"
              onClick={[Function]}
            >
              Configure
            </div>
            <div
              className="tabs__tab"
              onClick={[Function]}
            >
              Console
            </div>
            <div
              className="tabs__tab"
              onClick={[Function]}
            >
              History
            </div>
          </div>
        </div>
        <div
          className="tabs__applications-container"
        >
          <div
            className="configure tabs__aplication-container--show"
          >
            <div
              className="network-devices"
            >
              <div>
                <div
                  className="device-container-name"
                >
                  <svg
                    className="plus-circle"
                    data-testid="plus-icon"
                    onClick={[Function]}
                  >
                    add.svg
                  </svg>
                  Router
                </div>
                <div
                  className="device-container"
                >
                  <div
                    className="device-container-content"
                  >
                    <div />
                  </div>
                </div>
              </div>
              <div>
                <div
                  className="device-container-name"
                >
                  <svg
                    className="plus-circle"
                    data-testid="plus-icon"
                    onClick={[Function]}
                  >
                    add.svg
                  </svg>
                  Switch
                </div>
                <div
                  className="device-container"
                >
                  <div
                    className="device-container-content"
                  >
                    <div />
                  </div>
                </div>
              </div>
              <div>
                <div
                  className="device-container-name"
                >
                  <svg
                    className="plus-circle"
                    data-testid="plus-icon"
                    onClick={[Function]}
                  >
                    add.svg
                  </svg>
                  Host
                </div>
                <div
                  className="device-container"
                >
                  <div
                    className="device-container-content"
                  >
                    <div />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h2
            className="console tabs__aplication-container--hide"
            style={
              Object {
                "padding": "20px",
              }
            }
          >
            Coming soon...
          </h2>
          <h2
            className="console tabs__aplication-container--hide"
            style={
              Object {
                "padding": "20px",
              }
            }
          >
            Coming soon...
          </h2>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "#212529",
          "borderRadius": "10px",
          "boxShadow": "0 0 0 0.4vw #454950",
          "flex": "1 1 auto",
          "height": "57.15vh",
          "marginTop": "2.5vh",
          "overflow": "hidden",
          "position": "relative",
          "width": "68.18vw",
          "zIndex": 1,
        }
      }
    >
      <div
        className="icons"
      >
        <button
          className="iconButtons"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg>
            zoomInIcon.svg
          </svg>
        </button>
        <button
          className="iconButtons"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg>
            zoomOutIcon.svg
          </svg>
        </button>
        <button
          className="iconButtons"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg>
            hideIcon.svg
          </svg>
        </button>
        <button
          className="iconButtons"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg>
            centerImageIcon.svg
          </svg>
        </button>
      </div>
      <div
        className="handle react-draggable"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        style={
          Object {
            "backgroundColor": "#212529",
            "borderRadius": "10px",
            "cursor": "",
            "transform": "translate(0px,0px)",
            "visibility": "initial",
          }
        }
        transform={null}
      />
    </div>
  </div>
</div>
`;
